client/my-sites/stats/stats-purchase/stats-purchase-checkout-redirect.tsx

```diff
@@ -24,9 +24,13 @@ const getStatsCheckoutURL = (
 	adminUrl?: string,
 	isUpgrade?: boolean
 ) => {
-	const isFromJetpack = from?.startsWith( 'jetpack' );
-	// Get the checkout URL for the product, or the siteless checkout URL if from Jetpack or no siteSlug is provided
-	const checkoutType = ( isFromJetpack && ! isUpgrade ) || ! siteSlug ? 'jetpack' : siteSlug;
+	// Get the checkout URL for the product.
+	// The following cases get the logged out flow:
+	// - If the user is coming from My Jetpack
+	// - If no siteSlug is provided
+	// - If it's not an upgrade (currently causes errors in the logged in flow)
+	const useLoggedOutFlow = from === 'jetpack-my-jetpack' || from === 'jetpack-stats-upgrade-notice';
```

Comment:
The idea was to use the logged out cart in cases where it won't generate an error. In my testing, that means we can only use it if the site doesn't currently have a plan. Switching from personal to commercial should only happen for users without a plan, I think. Otherwise they wouldn't have been directed to the PWYW flow.

Users coming from the `jetpack-stats-upgrade-notice` are also unlikely to have a plan.